OpenRA.Mods.TS/Traits/Buildings/LowBridge.cs

```diff
@@ -24,27 +25,39 @@ class LowBridgeInfo : ITraitInfo, Requires<BuildingInfo>
 		public object Create(ActorInitializer init) { return new LowBridge(init.Self, this); }
 	}
 
-	class LowBridge : INotifyCreated
+	class LowBridge : INotifyAddedToWorld, INotifyRemovedFromWorld
 	{
 		readonly LowBridgeInfo info;
+		readonly IEnumerable<CPos> cells;
 
 		public LowBridge(Actor self, LowBridgeInfo info)
 		{
 			this.info = info;
+
+			var buildingInfo = self.Info.TraitInfo<BuildingInfo>();
+			cells = FootprintUtils.PathableTiles(self.Info.Name, buildingInfo, self.Location);
 		}
 
-		void INotifyCreated.Created(Actor self)
+		void UpdateTerrain(Actor self, byte terrainIndex)
 		{
-			var buildingInfo = self.Info.TraitInfo<BuildingInfo>();
-			var cells = FootprintUtils.PathableTiles(self.Info.Name, buildingInfo, self.Location);
-			var tileSet = self.World.Map.Rules.TileSet;
-			var terrainIndex = tileSet.GetTerrainIndex(info.TerrainType);
 			foreach (var cell in cells)
 				self.World.Map.CustomTerrain[cell] = terrainIndex;
 
 			var domainIndex = self.World.WorldActor.TraitOrDefault<DomainIndex>();
 			if (domainIndex != null)
 				domainIndex.UpdateCells(self.World, cells);
 		}
+
+		void INotifyAddedToWorld.AddedToWorld(Actor self)
+		{
+			var tileSet = self.World.Map.Rules.TileSet;
+			var terrainIndex = tileSet.GetTerrainIndex(info.TerrainType);
+			UpdateTerrain(self, terrainIndex);
+		}
+
+		void INotifyRemovedFromWorld.RemovedFromWorld(Actor self)
```

Comment:
Note that my last request was to tidy up a specific code issue with the trait so that we could merge the code at a logical and self-contained stopping point.  The extra changes that you added on top of this exposes a new problem which you could alternatively avoid here by removing them for a future PR.  So please don't go off on me about scope-creep, I'm just trying to avoid merging things that are obviously half-baked.